src/covering.ml

```diff
@@ -797,7 +799,7 @@ let env_of_rhs evars ctx env s lets =
     in
       pats, x @ y, List.length x 
   in
-  let pats = List.map (lift (-letslen)) pats @ map (lift len) patslets in
+  let pats = List.map (lift (-letslen)) pats @ patslets in
```

Comment:
It looks wrong after a cursory glance indeed, pats and patslets should live in the original ctx.